src/tgen-timer.c

```diff
@@ -73,9 +71,12 @@ tgentimer_settime_micros(TGenTimer *timer, guint64 micros)
     }
 }
 
-TGenEvent tgentimer_onEvent(TGenTimer* timer, gint descriptor, TGenEvent events) {
+TGenIOResponse tgentimer_onEvent(TGenTimer* timer, gint descriptor, TGenEvent events) {
     TGEN_ASSERT(timer);
 
+    TGenIOResponse response;
```

Comment:
Nit - instead of memset you can zero-initialize with:
TGenIOResponse response = {};

From ["struct initialization"](https://en.cppreference.com/w/c/language/struct_initialization) "All members that are not initialized explicitly are initialized implicitly the same way as objects that have static storage duration."

And from ["implicit initialization"](https://en.cppreference.com/w/c/language/initialization#Implicit_initialization) "objects with static and thread-local storage duration are initialized as follows..." <set to null or zero as appropriate, including padding>

Reply:
You can apparently also init char arrays as `char buf[10] = "";` :)

Reply:
> You can apparently also init char arrays as `char buf[10] = "";` :)

Huh, I don't think I've seen that idiom before, though I think I have seen `char buf[10] = {}`. Do you have a reference for `= ""`?

Also when looking myself for a reference, I stumbled across that technically an empty initializer list isn't allowed in C (for structs either). You can do = {0} for both structs and arrays, though. https://stackoverflow.com/a/17589839

Reply:
> Huh, I don't think I've seen that idiom before, though I think I have seen char buf[10] = {}. Do you have a reference for = ""?

My non-authoritative reference is [here](https://stackoverflow.com/questions/18688971/c-char-array-initialization).

> Also when looking myself for a reference, I stumbled across that technically an empty initializer list isn't allowed in C (for structs either). You can do = {0} for both structs and arrays, though.

:( I guess another force push is coming...